### PR TITLE
Fix: Right panel keeps showing chat when unmaximizing widget. 

### PR DESCRIPTION
**Pull request by manancodes**
## Checklist

- [x] Tests written for new code (and old code if feasible)
- [x] Linter and other CI checks pass
- [x] Sign-off given on the changes (see [CONTRIBUTING.md] (https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Fixes: https://github.com/vector-im/element-web/issues/26265
Signed-off-by: Manan Sadana manancodes.dev@gmail.com

This was the behaviour before the fix:
Unmaximizing the widget does not close the Right panel, resulting in 2 chat views
Before

And this is the behaviour after the fix:
Unmaximizing closes the Right panel.
After

Type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix: Right panel keeps showing chat when unmaximizing widget.  ([\#11697](https://github.com/matrix-org/matrix-react-sdk/pull/11697)). Fixes vector-im/element-web#26265. Contributed by @manancodes.

### CODE REVIEW

**Comment by t3chguy** [comment]
> Tests written for new code (and old code if feasible)

You checked this but actually wrote no tests..?

**Comment by manancodes** [comment]
> > Tests written for new code (and old code if feasible)
> 
> You checked this but actually wrote no tests..?

Sorry, corrected now. 
Did not realise I had to write tests for this. I'm still getting familiar with testing, looking into this. If you could refer some resources, it would really help.

**Comment by andybalaam** [comment]
> Did not realise I had to write tests for this. I'm still getting familiar with testing, looking into this. If you could refer some resources, it would really help.

There's some general guidance at https://github.com/matrix-org/matrix-react-sdk/blob/develop/docs/cypress.md

> Also, there must be tests which check if the chat opens in Right Panel when widget is maximised. Where can I find that? It will help me with writing the new tests. Thanks.

Take a look at `cypress/e2e/right-panel/right-panel.spec.ts` and the other tests in that directory. Hopefully they are similar to what you need. You might possibly need to add some new Cypress commands, similar to `cy.openRoomSettings();` which is fiddly but can be done. Take a look at how that one is made.

This might be quite difficult at first, so do please ask questions in #element-dev:matrix.org . It will be easiest to help if you have some code in a branch or PR that you can point at, so your questions are specific.

Thanks!

**Comment by manancodes** [comment]
I'm still learning cypress, I have some questions:

1. Am I using the right selectors?
2. I'm sure this is not enough, how can I check if the widget is taking the whole height when maximising?
3. It checks if the right panel is opened, how can I check if the right panel has the chat in it?
4. Are the tests in the right file? If no, where should I put them?

**Comment by manancodes** [comment]
@florianduros Can you please look into this?

**Comment by florianduros** [comment]
> @florianduros Can you please look into this?

I'll take a look today or tomorrow

**Review by florianduros — CHANGES_REQUESTED** [review]

File: cypress/e2e/widgets/layout.spec.ts [hunk 1]
```diff
@@ -129,4 +129,16 @@ describe("Widget Layout", () => {
 
         cy.get('iframe[title="widget"]').invoke("height").should("be.greaterThan", 400);
     });
+    it("open right panel with chat when maximizing widget", () => {
+        cy.get('iframe[title="widget"]').invoke("height").should("be.lessThan", 250);
+        cy.get('div[title="Maximise"]').click();
```
[the remark this answers is not in the document]
Comment (manancodes) [reply cypress/e2e/widgets/layout.spec.ts]: I have used accessible selectors now, can you review again? Thanks.

**Comment by manancodes** [comment]
Also, when you get time, can you please answer the queries that I had?

**Comment by manancodes** [comment]
Is it ready to be merged?
I'm participating in hacktoberfest so it will help if this gets merged soon. Thanks.

**Comment by florianduros** [comment]
> Also, when you get time, can you please answer the queries that I had?

1. With your last changes yes!
2/3. Let me check how this part of the code is made and your fix is working
4. They seems to be in the right file.

**Comment by florianduros** [comment]
@manancodes 

I think we can check if the chat/timeline is in the right panel with this piece of code:

```ts
this.context.rightPanelStore.currentCard.phase === RightPanelPhases.Timeline
```

**Comment by manancodes** [comment]
Thanks. I will add this.

**Comment by manancodes** [comment]
@florianduros 

This is how it is being checked in another file

https://github.com/matrix-org/matrix-react-sdk/blob/fd08e742757031bbb1ec7b050f83642d712c5fd0/cypress/e2e/right-panel/right-panel.spec.ts#L44-L46

How about using
`cy.get(".mx_BaseCard_header").should("contain", "Chat"); `

I am new to cypress, implementing your suggestion will take longer because I could not find a similar implementation. I will work on this if you are not ok with my proposal.

**Comment by florianduros** [comment]
@manancodes I thought you asked how to check in the component implementation, not in the test. My fault.

`cy.get(".mx_BaseCard_header").should("contain", "Chat"); ` is not specific to the right panel, I think it will get the main chat too. 

We need first to select the right panel and query the chat in it:

```ts
cy.get(".mx_RightPanel").within(() => {
    cy.get(".mx_BaseCard_header").should("contain", "Chat"));
})
```
[Cypress documentation within](https://docs.cypress.io/api/commands/within)

**Comment by manancodes** [comment]
Perfect, I will add this.

**Comment by manancodes** [comment]
Done. Added and tested.

**Comment by manancodes** [comment]
@florianduros , sorry for disturbing you, but I'm participating in hacktoberfest, if this get's merged soon it will really help me with that.

**Review by florianduros — APPROVED** [review]
Thanks!

**Pull request merged**